File: testdata/testkit/python/domain/entity.py
"""Core domain entities."""
from dataclasses import dataclass
from typing import Optional, Protocol


@dataclass
class Entity:
    id: str
    name: str
    data: dict


class Repository(Protocol):
    """Port interface for data access."""
    def find_by_id(self, id: str) -> Optional[Entity]: ...
    def save(self, entity: Entity) -> None: ...


class Service:
    """Domain service orchestrating business logic."""
    def __init__(self, repo: Repository):
        self._repo = repo

    def get_entity(self, id: str) -> Optional[Entity]:
        return self._repo.find_by_id(id)

File: testdata/testkit/go/domain/entity.go
package domain

// Entity is the core domain type.
type Entity struct {
	ID   string
	Name string
	Data map[string]string
}

// Repository is the port interface for data access.
type Repository interface {
	FindByID(id string) (*Entity, error)
	Save(entity *Entity) error
}

// Service orchestrates domain logic.
type Service struct {
	repo Repository
}

// NewService creates a new Service.
func NewService(repo Repository) *Service {
	return &Service{repo: repo}
}

// GetEntity retrieves an entity by ID.
func (s *Service) GetEntity(id string) (*Entity, error) {
	return s.repo.FindByID(id)
}
